Sources/Common/Background Queue/QueueStorage.swift

```diff
@@ -24,99 +24,76 @@ import Foundation
-    func delete(storageId: String) -> Bool
-    func deleteExpired() -> [QueueTaskMetadata]
+    func getInventory(siteId: String) -> [QueueTaskMetadata]
+    func get(storageId: String, siteId: String) -> QueueTask?
+    func delete(storageId: String, siteId: String) -> Bool
 }
 
 // sourcery: InjectRegister = "QueueStorage"
 public class FileManagerQueueStorage: QueueStorage {
     private let fileStorage: FileStorage
     private let jsonAdapter: JsonAdapter
-    let siteId: String
-    private let sdkConfig: SdkConfig
     let logger: Logger
     let dateUtil: DateUtil
     private var inventoryStore: QueueInventoryMemoryStore
 
     let lock: Lock
 
-    private var inventory: [QueueTaskMetadata]? {
-        get {
-            lock.lock()
-            defer { lock.unlock() }
-
-            if let inventoryCache = inventoryStore.inventory {
-                return inventoryCache
-            }
-
-            guard let data = fileStorage.get(type: .queueInventory, fileId: nil) else { return nil }
-            guard let readInventory: [QueueTaskMetadata] = jsonAdapter.fromJson(data) else { return nil }
-            inventoryStore.inventory = readInventory // set in-memory cache for next time getter is called
-
-            return readInventory
-        }
-        set {
-            lock.lock()
-            defer { lock.unlock() }
-
-            guard let data = jsonAdapter.toJson(newValue) else {
-                return
-            }
-
-            // the inventory is the BQ's single source of truth for what tasks are in the BQ. It's important that the inventory cache reflects what's in SDK storage so only update
-            // it after we successfully save the storage.
-            // If there is a failed save to file system, the item added to the BQ will get ignored to try and keep the SDK into an error-free state.
-            let successfullySavedInStorage = fileStorage.save(type: .queueInventory, contents: data, fileId: nil)
-
-            if successfullySavedInStorage {
-                inventoryStore.inventory = newValue // update cache
-            }
-        }
-    }
-
     init(
         fileStorage: FileStorage,
         jsonAdapter: JsonAdapter,
         lockManager: LockManager,
-        sdkConfig: SdkConfig,
         logger: Logger,
         dateUtil: DateUtil,
         inventoryStore: QueueInventoryMemoryStore
     ) {
-        self.siteId = sdkConfig.siteId
         self.fileStorage = fileStorage
         self.jsonAdapter = jsonAdapter
-        self.sdkConfig = sdkConfig
         self.logger = logger
         self.dateUtil = dateUtil
         self.lock = lockManager.getLock(id: .queueStorage)
         self.inventoryStore = inventoryStore
     }
 
-    public func getInventory() -> [QueueTaskMetadata] {
-        inventory ?? []
+    public func getInventory(siteId: String) -> [QueueTaskMetadata] {
+        lock.lock()
+        defer { lock.unlock() }
+
+        if let inventoryCache = inventoryStore.inventory {
```

Comment:
maybe we can get rid memory inventory as well? as we just need to do this operation once?

Reply:
I considered that, but I don't think that we should because the BQ migration code would take a performance hit. Since the BQ migration code does both reading and writing to the BQ storage, the in-memory cache is providing value.